mobile/src/screens/note/note.tsx

```diff
@@ -0,0 +1,153 @@
+import { type RouteProp, useRoute } from '@react-navigation/native';
+import React from 'react';
+
+import {
+  BackHandler,
+  FormController,
+  LinearGradient,
+  ScrollView,
+  TextInput,
+  View,
+} from '#libs/components/components';
+import { debounce, sanitizeInput } from '#libs/helpers/helpers';
+import {
+  useAppDispatch,
+  useAppForm,
+  useAppSelector,
+  useCallback,
+  useEffect,
+  useMemo,
+} from '#libs/hooks/hooks';
+import { type JournalNavigationParameterList } from '#libs/types/types';
+import { type JournalEntryCreateRequestDto } from '#packages/journal/journal';
+import { actions as journalActions } from '#slices/journal/journal';
+
+import {
+  DEFAULT_NOTE_PAYLOAD,
+  NOTE_SANITIZER_OPTIONS,
+  SAVE_NOTE_TIMEOUT,
+  TEXT_PLACEHOLDER,
+} from './libs/constants';
+import { styles } from './styles';
+
+type NoteScreenRouteProperty = RouteProp<
+  JournalNavigationParameterList,
+  'Note'
+>;
+
+const handleSanitizeInput = (textInput: string): string => {
+  return sanitizeInput(textInput, NOTE_SANITIZER_OPTIONS);
+};
+
+const Note: React.FC = () => {
+  const route = useRoute<NoteScreenRouteProperty>();
+  const id = route.params.id;
+
+  const dispatch = useAppDispatch();
+
+  const { selectedJournalEntry } = useAppSelector(({ journal }) => {
+    return {
+      selectedJournalEntry: journal.selectedJournalEntry,
+    };
+  });
+
+  const { control, watch, isDirty } = useAppForm({
+    defaultValues: selectedJournalEntry
+      ? {
+          title: selectedJournalEntry.title || DEFAULT_NOTE_PAYLOAD.title,
+          text: selectedJournalEntry.text || DEFAULT_NOTE_PAYLOAD.text,
+        }
+      : DEFAULT_NOTE_PAYLOAD,
```

Comment:
Maybe you can create a helper to get default values

Reply:
@fshabanov this functionality is crucial for correct note displaying (existing and new) and i would better leave it here and dont change anything